model/commitq/merge_function_registry.go

```diff
@@ -0,0 +1,31 @@
+package commitq
+
+import (
+	"github.com/evergreen-ci/evergreen/thirdparty"
+)
+
+type MergeFunction interface{}
```

Comment:
Will we want to pass something radically different from title, message, and sha? Although we may support something other than GitHub, ultimately we are merging git commits, which have a finite set of properties. I think you could define a struct with those arguments and have a function type here instead of an empty interface.

Reply:
Done.